compat/v1/compat.go

```diff
@@ -322,22 +322,14 @@ func QueryStateResponseFromState(state map[string]engine.XState) (QueryStateResp
 	return QueryStateResponse{State: stateResponse}, nil
 }
 
-type SuccessfulOutput struct {
-	Hash    pack.Bytes32 `json:"hash"`
-	Amount  pack.U256    `json:"amount"`
-	Sighash pack.Bytes32 `json:"sighash"`
-	Sig     pack.Bytes65 `json:"sig"`
-	Txid    pack.Bytes   `json:"txid"`
-}
-
 // v0.4 darknodes respond with empty strings instead of omitting nil fields
 // This causes issues with ren-js v2 so we need to manually strip the responses
-func TxOutputFromV2QueryTxOutput(output engine.LockMintBurnReleaseOutput) SuccessfulOutput {
-	return SuccessfulOutput{
-		Hash:    output.Hash,
-		Amount:  output.Amount,
-		Sighash: output.Sighash,
-		Sig:     output.Sig,
-		Txid:    output.Txid,
-	}
+func TxOutputFromV2QueryTxOutput(output engine.LockMintBurnReleaseOutput) pack.Typed {
+	return pack.NewTyped(
+		"hash", output.Hash,
+		"amount", output.Amount,
+		"sighash", output.Sighash,
+		"sig", output.Sig,
+		"txid", output.Txid,
```

Comment:
There's a new `Txindex` field that was missing previously, so you might need to set it here.